dashboard-api/internal/platform/server/middleware/auth/auth.go

```diff
@@ -0,0 +1,65 @@
+package auth
+
+import (
+	"log"
+	"net/http"
+	"net/url"
+	"time"
+
+	jwtmiddleware "github.com/auth0/go-jwt-middleware"
+	"github.com/auth0/go-jwt-middleware/validate/josev2"
+	"github.com/gin-gonic/gin"
+	"gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+func EnsureValidTokenMiddleware(domain, audience string) gin.HandlerFunc {
+	issuerURL, err := url.Parse(domain)
+	if err != nil {
+		log.Fatalf("EnsureValidTokenMiddleware: failed to parse the issuer url: %v", err)
+	}
+
+	keyFunc := josev2.NewCachingJWKSProvider(*issuerURL, 5*time.Minute).KeyFunc
+	expectedClaimsFunc := func() jwt.Expected {
+		return jwt.Expected{
+			Issuer:   domain,
+			Audience: []string{audience},
+		}
+	}
+
+	validator, err := josev2.New(
+		keyFunc,
+		jose.RS256,
+		josev2.WithExpectedClaims(expectedClaimsFunc),
+	)
+	if err != nil {
+		log.Fatalf("EnsureValidTokenMiddleware: failed to set up the josev2 validator: %v", err)
+	}
+
+	/*
+		This error handler func is required to override the default behaviour of the jwtmiddleware
+		which writes to the headers and body of the response upon error.
+	*/
+	noopErrorHandlerFunc := func(w http.ResponseWriter, r *http.Request, err error) {}
+	middleware := jwtmiddleware.New(validator.ValidateToken, jwtmiddleware.WithErrorHandler(noopErrorHandlerFunc))
+
+	return func(ctx *gin.Context) {
+		var encounteredError = true
+
+		var handler http.HandlerFunc = func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I'm getting a bit of bad code smell here, but I don't know the library. 

It feels a bit like they give you a middleware, but you don't use it so you create your own. That's fine. But in that case it's probably more straight-forward to use some lower-level methods, if they give it to you, to just check the JWT. 

In other words, I'd expect something like: 

 ```go
 ok, err :=  validator.ValidateToken(tokenFromCtxSomehow)
if err != nil {
  ctx.AbortWithStatusJSON(...)
}
ctx.Next()
 ```

Would be sufficient. What else is this providing?

Reply:
(that's psuedo-code, I have no idea what the API is of validator!